components/layout_2020/flow/float.rs

```diff
@@ -914,43 +914,61 @@ impl FloatBox {
                         // Calculate inline size.
                         // https://drafts.csswg.org/css2/#float-width
                         let style = non_replaced.style.clone();
-                        let box_size = style
-                            .content_box_size_deprecated(containing_block, &pbm)
-                            .map(|v| v.map(Au::from));
-                        let max_box_size = style
-                            .content_max_box_size_deprecated(containing_block, &pbm)
-                            .map(|v| v.map(Au::from));
-                        let min_box_size = style
-                            .content_min_box_size_deprecated(containing_block, &pbm)
-                            .map(|v| v.map(Au::from))
-                            .auto_is(Au::zero);
-
-                        let block_size = box_size.block.map(|size| {
-                            size.clamp_between_extremums(min_box_size.block, max_box_size.block)
-                        });
-                        let tentative_inline_size = box_size.inline.auto_is(|| {
-                            let available_size =
-                                containing_block.inline_size - pbm_sums.inline_sum();
+                        let box_size = style.content_box_size(containing_block, &pbm);
+                        let max_box_size = style.content_max_box_size(containing_block, &pbm);
+                        let min_box_size = style.content_min_box_size(containing_block, &pbm);
+                        let available_inline_size =
+                            containing_block.inline_size - pbm_sums.inline_sum();
```

Comment:
Should this take into account other floats that might be taking up inline space or do we first possibly stretch and then place among floats?

Reply:
No, floats receive the full containing block, just like inline-blocks, see https://github.com/w3c/csswg-drafts/issues/4028#issuecomment-2374038995